employees/forms.py

```diff
@@ -89,41 +89,35 @@ def __init__(self, *args: Any, **kwargs: Any):
-            with suppress(ValueError, TypeError):
-                project_id = int(self.data.get("project"))
-                self.fields["task_activities"].queryset = TaskActivityType.objects.filter(projects=project_id).order_by(
-                    "name"
-                )
-        elif self.instance.pk:
-            self.fields["task_activities"].queryset = (
-                self.instance.project.project_activities.order_by("name")
-                | TaskActivityType.objects.filter(pk=self.instance.task_activities.pk)
-            ).distinct()
+        self._set_choices_for_fields_in_report_from(author)
 
-    def _set_last_choices_in_report_form(self, author: CustomUser) -> None:
+    def _set_choices_for_fields_in_report_from(self, author: CustomUser) -> None:
         if self.instance.pk is None:
-            last_report = author.report_set.order_by("-creation_date").first()
-            if last_report and (
-                author in last_report.project.members.all() or author in last_report.project.managers.all()
-            ):
-                self.initial["project"] = last_report.project
-                self.fields["task_activities"].queryset = last_report.project.project_activities.all()
-                self.initial["task_activities"] = last_report.task_activities
-            elif self.fields["project"].queryset.exists():
-                self.initial["project"] = self.fields["project"].queryset.first()
-                self.fields["task_activities"].queryset = self.initial["project"].project_activities.all()
-            else:
-                self.fields["task_activities"].queryset = TaskActivityType.objects.none()
+            self._set_last_choices_while_creating_new_report_form(author)
         else:
-            self.fields["project"].queryset = (
-                self.fields["project"].queryset | Project.objects.filter(pk=self.instance.project.pk).distinct()
-            )
+            self._set_choices_from_edited_report()
+
+    def _set_last_choices_while_creating_new_report_form(self, author: CustomUser) -> None:
+        last_report = author.report_set.order_by("-creation_date").first()
+        if last_report and (
+            author in last_report.project.members.all() or author in last_report.project.managers.all()
+        ):
+            self.initial["project"] = last_report.project
+            self.initial["task_activity"] = last_report.task_activity
+        elif self.fields["project"].queryset.exists():
+            self.initial["project"] = self.fields["project"].queryset.first()
+        self.fields["task_activity"].queryset = TaskActivityType.objects.get_active()
+
+    def _set_choices_from_edited_report(self) -> None:
+        current_report = get_object_or_404(Report, pk=self.instance.pk)
+        self.initial["project"] = current_report.project
+        self.fields["project"].queryset = (
+            self.fields["project"].queryset | Project.objects.filter(pk=self.instance.project.pk).distinct()
```

Comment:
You don't need `distinct` method here

Reply:
Cannot remove `distinct`. Without it django cannot combine two queries. AssertionError: Cannot combine a unique query with a non-unique query.